tests/core/policies/test_memoization.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from tests.core.test_policies import PolicyTestCollection
+from typing import Optional, Dict, Text, Any
+from rasa.core.featurizers.tracker_featurizers import (
+    TrackerFeaturizer,
+    MaxHistoryTrackerFeaturizer,
+)
+from rasa.shared.core.generator import TrackerWithCachedStates
+from rasa.core.policies.memoization import AugmentedMemoizationPolicy, MemoizationPolicy
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.events import (
+    ActionExecuted,
+    UserUttered,
+    SlotSet,
+)
+from rasa.shared.nlu.interpreter import RegexInterpreter
+
+
+class TestMemoizationPolicy(PolicyTestCollection):
```

Comment:
@twerkmeister I think you might need to do some changes here as well in case that is merged before your adaption PR. I've already done some adaptions here to make the CI pass